python/lsst/ts/standardscripts/maintel/m1m3/check_actuators.py

```diff
@@ -0,0 +1,226 @@
+# This file is part of ts_standardscripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["CheckActuators"]
+
+
+import time
+import yaml
+import asyncio
+
+from lsst.ts.idl.enums import MTM1M3
+from lsst.ts.idl.enums.Script import ScriptState
+from lsst.ts.observatory.control.maintel.mtcs import MTCS, MTCSUsages
+
+import lsst.ts.cRIOpy.M1M3FATable as M1M3FATable
+from lsst.ts.cRIOpy.M1M3FATable import FATABLE, FATABLE_TYPE
+
+from lsst.ts.salobj import BaseScript
+
+
+class CheckActuators(BaseScript):
+    """Run a M1M3 bump test on individual actuators.
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+
+    Notes
+    -----
+    **Checkpoints**
+
+    - "Testing hardpoint {id}.": Check hardpoint.
+    - "Hardpoint breakaway check complete.": Check complete.
+
+    """
+
+    def __init__(self, index, add_remotes: bool = True):
+        super().__init__(index=index, descr="Bump Test on M1M3 Actuators")
+
+        self.mtcs = MTCS(
+            domain=self.domain,
+            intended_usage=None if add_remotes else MTCSUsages.DryTest,
+            log=self.log,
+        )
+
+        # Average time for a bump test is 23s
+        self.timeout_one_bump = 25
+
+        # Getting ids from mtcs
+        self.m1m3_actuator_ids = self.mtcs.get_m1m3_actuator_ids()
+        self.m1m3_secondary_actuator_ids = self.mtcs.get_m1m3_actuator_secondary_ids()
+
+        self.total_actuators = len(self.m1m3_actuator_ids)
+        self.total_sec_actuators = len(self.m1m3_secondary_actuator_ids)
+
+        # Getting actuators ids
+        self.actuators = self.m1m3_actuator_ids
+
+    async def assert_feasibility(self):
+        """Verify that the system is in a feasible state before
+        running bump test.
+        """
+
+        for comp in self.mtcs.components_attr:
+            if comp != "mtm1m3":
+                self.log.debug(f"Ignoring component {comp}.")
+                setattr(self.mtcs.check, comp, False)
+
+        await asyncio.gather(
+            self.mtcs.assert_all_enabled(),
+            self.mtcs.assert_liveliness(),
+        )
```

Comment:
Do you know if the m1m3 must be parked for this operations? if so, consider including

```
self.mtcs.assert_m1m3_detailed_state(
                {DetailedState.PARKED, DetailedState.PARKEDENGINEERING}
            )
```

here.

Reply:
Ohhh... I just saw this after answering a similar question below. The original idea of this script was to test it either in `DetailedState.PARKEDENGINEERING` or `DetailedState.ACTIVEENGINEERING` but without changing the detailed state. However, the reason for that is to avoid script duplication. That could be avoided by modifying the JSON schema to include those options, right?

Reply:
In any case, I'm adding the following to the code:

```
self.mtcs.assert_m1m3_detailed_state(
            {
                DetailedState.PARKED,
                DetailedState.PARKEDENGINEERING,
                DetailedState.ACTIVE,
                DetailedState.ACTIVEENGINEERING,
            }
        )
```
Is this correct?

Also: flake8 is complaining about the detailed states, saying e.g. that the `DetailedState` name is undefined. Should I import something?

Reply:
yes, you need to import it:

```
from lsst.ts.idl.enums.MTM1M3 import DetailedState
```